lib/metrics/index.js

```diff
@@ -0,0 +1,144 @@
+            return client.constructor.name === name;
+        });
+    }
+
+// Example Options:
+// {
+//     type: 'Counter',
+//     name: 'hitcount',
+//     prometheus: {
+//         name: 'hitcount',
+//         help: 'hit count',
+//         buckets: [], // https://github.com/siimon/prom-client#histogram
+//         percentiles: [], // https://github.com/siimon/prom-client#summary
+//     },
+//     labels: {
+//         keys: [],
+//         labelPosition: 'before',
+//         omitLabelNames: false
+//     }
+// }
+    makeMetric(options) {
+        const metric = new Metric(this.clients, this.logger, options);
+        this.cache.set(options.name, metric);
+        return metric;
+    }
+
+    fetchMetric(name) {
+        return this.cache.get(name);
+    }
+
+    close() {
+        this.clients.some((o) => o.close());
+        this.clients = [];
+    }
+}
+
+// To preserve backwards compatibility, a subset of statsd and log clients
+// methods are mapped into the proxy interface and a warning logged.
+// Also log attempts to use unknown/unmapped functions.
+// Example Options:
+// [
+//     {
+//         type: 'log',
+//         name: 'service-runner'
+//     },
+//     {
+//         type: 'statsd',
+//         host: 'localhost',
+//         port: '8125',
+//         name: 'service-runner'
+//     },
+//     {
+//         type: 'prometheus',
+//         port: 9000,
+//         name: 'service-runner'
+//     }
+// ]
+function makeMetrics(options, logger) {
+    const metrics = new Metrics(options, logger);
+    const handler = {
+        get: function (target, prop) {
+            if (target[prop] === undefined) {
+                logger.log('error/metrics', `No such method '${prop.toString()}' in Metrics`);
+                return function () {};
+            } else {
+                return target[prop];
+            }
+        }
+    };
+    const proxy = new Proxy(metrics, handler);
+
+    // The metrics interface exposed the statsd clients directly
+    // and many dependent applications use these functions directly.
+    // Build up a backwards-compatible interface by mapping these functions
+    // to the Metrics proxy instance and warn on deprecated usage.
+    DEPRECATED_METHODS.forEach((o) => {
+        proxy[o] = function (...args) {
+            logger.log('warn/metrics', `Calling 'Metrics.${o}' directly is deprecated.`);
+            metrics.clients.forEach((client) => {
+                if (client.constructor.name !== 'PrometheusClient') {
+                    client.client[o].apply(client.client, args);
+                }
+            });
+        };
+    });
+
+    // Deprecated.
+    // Support creating sub-metric clients with a fixed prefix. This is useful
+    // for systematically categorizing metrics per-request, by using a
+    // specific logger.  WARNING: if both Log and StatsD are configured, only
+    // one will be returned.
+    proxy.makeChild = function (name) {
```

Comment:
So, what's a replacement for this in a new paradigm?

Reply:
Yes or using the same metric with a different label value.  More information about the Prometheus data model can be found [here](https://prometheus.io/docs/concepts/data_model/).

In short, the metric names are more static and made multi-dimensional through "tagging" with labels of varying values.